src/cryptoadvance/specter/templates/wallet_send_sign_psbt.html

```diff
@@ -0,0 +1,335 @@
+					document.getElementById("hwi_sign_btn_"+currentSigningDeviceNum).innerHTML += ' (&#10004;)';
+
+				}
+				// console.log(this.responseText);
+			}else{
+				// document.getElementById("lbl").innerHTML = "Connection failed... Trying again...";
+			}
+		}
+		xmlHttp.onerror = function(e){
+			console.log(xmlHttp);
+			// err = xmlHttp;
+		}
+		xmlHttp.open("POST", url, true); // true for asynchronous 
+		// xmlHttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+		xmlHttp.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+		xmlHttp.send(JSON.stringify({"psbt0": psbt0, "psbt1": psbt1, "txid": "{{psbt['tx']['txid']}}", "device_name": currentSigningDevice}));
+		// xmlHttp.send(null);
+	}
+	document.addEventListener("DOMContentLoaded", function(){
+		let el = document.getElementById("pastetx");
+		el.addEventListener("click", function(){
+			let input = prompt("Paste signed transaction here");
+			if(input){
+				combine(input);
+			}
+		});
+	});
+
+	{# ================== Signers various device support ================== #}
+	{% if wallet.uses_hwi_device %}
+		document.addEventListener("DOMContentLoaded", function(){
+			initPageOverlay("hwi_sign_container");
+
+			{% if not wallet.is_multisig %}
+				document.getElementById("hwi_sign_btn").addEventListener("click", async function(e) {
+					e.preventDefault();
+					beginDetectWallet("{{ wallet.device_type }}");
+				});
+			{% else %}
+				{% for device in wallet.devices %}
+					{% if device.type in ['coldcard', 'keepkey', 'ledger', 'trezor', 'specter'] %}
+						document.getElementById("hwi_sign_btn_{{ loop.index }}").addEventListener("click", async function(e) {
+							currentSigningDevice = "{{ device.name }}"
+							currentSigningDeviceNum = "{{ loop.index }}"
+							e.preventDefault();
+							beginDetectWallet("{{ device.type }}");
+						});
+					{% endif %}
+				{% endfor %}
+			{% endif %}
+
+			document.getElementById("hwi_include__unlocked__submit").addEventListener("click", async function(e) {
+				e.preventDefault();
+				document.getElementById("hwi_device_name").textContent = capitalize(hwiCurrentWallet.type);
+				showPageOverlay("hwi_sign_container");
+
+				const psbt = "{{ psbt['base64'] }}";
+				var jsonResponse = await hwiSendPsbt(psbt);
+				if (jsonResponse.hasOwnProperty('error')) {
+					alert(jsonResponse.error);
+				} else if (jsonResponse.psbt !== "undefined") {
+					combine(jsonResponse.psbt);
+				}
+
+				hidePageOverlay("hwi_sign_container");
+			});
+		});
+	{% endif %}
+
+
+	{% if wallet.uses_qrcode_device %}
+		import QrScanner from "/static/qr/qr-scanner.min.js";
+		QrScanner.WORKER_PATH = '/static/qr/qr-scanner-worker.min.js';
+
+		const video = document.getElementById('qr-video');
+
+		const scanner = new QrScanner(video, result => {
+			scanner.stop();
+			document.getElementById("popup").style.display = 'none';
+			{% if not psbt %}
```

Comment:
This part is used to scan destination address
So you need to copy QR scanning functionality to `wallet_send.html` as well
If we are in the send page - scanner should populate "address" field, if in PSBT - it should call combine function.

Reply:
Good spot! Right now i don't have a Hardware-wallet to test the PSBT-part but the destination-address is now working. Might do that tonight or tomorrow.